schema/validator.go

```diff
@@ -26,14 +26,42 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
+type validateRefMediaFunc func(r io.Reader) error
+
+var vlidateRefMediaMap = map[Validator]validateRefMediaFunc{
+	ValidatorMediaTypeImageIndex: validateImageIndexRefObject,
+	ValidatorMediaTypeManifest:   validateManifestRefObject,
+}
+
+// ValidateFunc provides the type of function to validate specific object.
+type ValidateFunc func(r io.Reader, v Validator) error
```

Comment:
> ValidateRefFuncs are public functions.

That doesn't really matter.  We could always provide `io.Reader` and parsed-JSON flavored versions if we figure out how to write a function that takes an already-parsed-from-JSON object.